src/api.ts

```diff
@@ -0,0 +1,70 @@
+import { request } from "./common"
+
+export type SpawnRequestBody = {
+    env?: Record<string, string>; // env vars always map strings to strings
+    port?: number;
+    tag?: string;
+}
+
+// TODO(paul): formalize the API return types and create a public API spec (OpenAPI?)
+
+export class JamsocketApi {
+    apiBase: string
+
+    constructor(private auth: string) {
+        this.apiBase = process.env.JAMSOCKET_SERVER_API ?? 'https://jamsocket.dev'
+    }
+
+    private async makeAuthenticatedRequest(endpoint: string, body?: any): Promise<any> {
+        const url = `${this.apiBase}${endpoint}`;
+        const response = await request(url, body || null, {
+            method: body ? 'POST' : 'GET',
```

Comment:
Is this always the case? that if no `body` is provided it should be a `GET`? Both the spawn request and the create spawn token request only optionally take the params in the body but are expected to be `POST`s

Reply:
Good point, fixed.